lms/static/js/header/header.js

```diff
@@ -0,0 +1,172 @@
+        if (isNext && !isHamburgerMenu && !isToggle && isLastItem) {
+            // Loop to the start from the final element
+            nextLink = isDropdownOption ? $toggleUserDropdown : $hamburgerMenu;
+        } else if (!isNext && (isHamburgerMenu || isToggle)) {
+            // Loop to the end when up arrow pressed from menu icon
+            nextLink = isHamburgerMenu ? $('.mobile-menu .mobile-nav-link a').last()
+                : $('.dropdown-user-menu .dropdown-nav-item').last().find('a');
+        } else if (isNext && (isHamburgerMenu || isToggle)) {
+            // Loop to the first element from the menu icon
+            nextLink = isHamburgerMenu ? $('.mobile-menu .mobile-nav-link a').first()
+                : $('.dropdown-user-menu .dropdown-nav-item').first().find('a');
+        } else {
+            // Loop up to the menu icon if first element in menu
+            if (!isNext && $(e.target).parent().is(':first-child') && !isHamburgerMenu && !isToggle) {
+                nextLink = isDropdownOption ? $toggleUserDropdown : $hamburgerMenu;
+            } else {
+                nextLink = isNext ?
+                    $(e.target).parent().next().find('a') : // eslint-disable-line newline-per-chained-call
+                    $(e.target).parent().prev().find('a'); // eslint-disable-line newline-per-chained-call
+            }
+        }
+        nextLink.focus();
+
+        // Don't let the screen scroll on navigation
+        e.preventDefault();
+        e.stopPropagation();
+    }
+
+    // Escape clears out of the menu
+    if (e.keyCode === 27 && (isDropdownOption || isHamburgerMenu || isMobileOption || isToggle)) {
+        if (isDropdownOption || isToggle) {
+            $('.global-header .nav-item .dropdown-user-menu').hide();
+            $toggleUserDropdown.focus();
+            $userMenu.attr('aria-expanded', 'false');
+            $userMenu.removeClass('open');
+        } else {
+            $('.mobile-menu').hide();
+            $hamburgerMenu.focus();
+            $hamburgerMenu.attr('aria-expanded', 'false');
+            $hamburgerMenu.removeClass('open');
+        }
+    }
+
+    // Loop when tabbing and using arrows
+    if ((e.keyCode === 9) && ((isDropdownOption && isLastItem) || (isMobileOption && isLastItem) || (isHamburgerMenu
+        && $hamburgerMenu.hasClass('open')) || (isToggle && $toggleUserDropdown.hasClass('open')))) {
+        nextLink = null;
+        loopFirst = isLastItem && !e.shiftKey && !isHamburgerMenu && !isToggle;
+        loopLast = (isHamburgerMenu || isToggle) && e.shiftKey;
+        if (!(loopFirst || loopLast)) {
+            return;
+        }
+        if (isDropdownOption || isToggle) {
+            nextLink = loopFirst ? $toggleUserDropdown : $('.dropdown-user-menu .dropdown-nav-item a').last();
+        } else {
+            nextLink = loopFirst ? $hamburgerMenu : $('.mobile-menu .mobile-nav-link a').last();
+        }
+        nextLink.focus();
+        e.preventDefault();
```

Comment:
Move this to the top of the if statement wrapper, this may be what's causing the inconsistent behaviour.